### DISCUSSION TITLE
Transaction cost is always higher than the testnet provided amount

### DISCUSSION THREAD

**Discussion by kppavlov** [Cyfrin/foundry-full-course-cu#3075] [Q&A]
I successfully got 0.05Eth on Sepolia test net. Successfully connected my wallet to Remix and I can see that my account has 0.05Eth. What I do not understand is why every time I want to deploy a contract the gas fee is greater than the amount I have in the account. Why does the faucet gives only 0.05Eth when the contract deployment cost more than that? Am I doing something wrong or I have to drain the faucet for a couple of days, so that I can have 1-2 tries of deployment on test net? Could someone suggest a solution to that?

**Comment by EngrPips** [comment]
Hello @kppavlov, The sepolia testnet has become more expensive to use lately, You should try to claim more sepolia eth and attempt the deployment or test things locally.

**Reply by kppavlov** [reply]
Thank you very much for the quick reply! Did not know there is an option to test locally. Is there a repo you would suggest which I can use to spin up a local chain?

**Reply by EngrPips** [reply]
Patrick did not talk about `Anvil` yet in the course, or are you not following Patrick's course on Cyfrin Updraft?

**Reply by kppavlov** [reply]
I am following the course but there was not a single word about that so far. Anyway I found one that is called Hardhat. It is a local blockchain I can use to deploy and test smart contracts. Thank you!

**Reply by EngrPips** [reply]
What lesson are you currently watching?

**Reply by kppavlov** [reply]
Solidity Smart Contract Development: Lesson 12: Deploying your first contract

**Reply by EngrPips** [reply]
Link to the lesson please.

**Reply by kppavlov** [reply]
This is the link to the whole course https://updraft.cyfrin.io/courses/solidity/simple-storage/deploying-solidity-smart-contract. I am at lesson 12

**Reply by EngrPips** [reply]
I see you are writing code in `Remix` and not in a local setup. You can deploy to `RemixVm` for now and skip the test net part if you are having a hard time getting enough test net funds. Patrick will later teach you how to deploy to a local Anvil chain and even create mock contracts to be able to do all your building and testing locally.
